### PR TITLE
Lvnod 47 auth

### PR DESCRIPTION
**Pull request by AlbaMontis**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-46)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-76)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-48)

## Code reviewers

- [x] @yurii-kokhalevych
- [x] @denisbodnar
- [x] @shyyka

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

### CODE REVIEW

File: app/src/helper/constants.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const API = {
+  HOST: 'http://localhost:',
+  PORT: '8082'
+}
+
+export { API };
```
Comment (denisbodnar) [inline app/src/helper/constants.js:6]: export API;

**Review by denisbodnar — DISMISSED** [review]

File: app/src/redux/Login.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  let isToken;
+  sessionStorage.getItem('token')? isToken = true:
+    isToken = false;
+  return {
+    type: CHECK_TOKEN,
+    isToken: isToken
+  }
+}
+
+export function logout(dispatch) {
```
Comment (AlbaMontis) [inline app/src/redux/Login.js]: Delete dispatch

File: app/src/store/configureStore.js [hunk 3]
```diff
@@ -1,14 +1,19 @@
 import { createStore, applyMiddleware } from 'redux';
-import { routerMiddleware } from 'react-router-redux';
+import { routerMiddleware, push } from 'react-router-redux';
```
Comment (AlbaMontis) [inline app/src/store/configureStore.js]: Delete push

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

File: app/src/helper/constants.js (hunk 1 above)
Comment (AlbaMontis) [reply app/src/helper/constants.js:6]: It could be `export const API`, but not `export API`

File: app/src/redux/reducer.js [hunk 4]
```diff
@@ -1,8 +1,10 @@
 import { combineReducers } from 'redux';
 import items from './Test';
+import login from './Login'
```
Comment (denisbodnar) [inline app/src/redux/reducer.js]: ;

**Review by denisbodnar — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

File: app/src/redux/Login.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+const initialState = {
+  illegalInput: false,
+  isToken: false
+};
+
+export default function loginReduser(state = initialState, action) {
+  switch (action.type) {
+    case LOGIN_USER_FAILURE:
+      return Object.assign({}, {
```
Comment (4asof4ik) [inline app/src/redux/Login.js]: You can return without object.assign

File: app/src/redux/Login.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    isToken: false
+  }
+}
+
+export function loginUser(email, password) {
+  return dispatch => {
+    dispatch(loginUserRequest());
+    request
+      .post(API.HOST + API.PORT + '/api/auth/login')
+      .send({email: email, password: password})
```
Comment (4asof4ik) [inline app/src/redux/Login.js]: Create let user with params

File: app/src/redux/Login.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+export function loginUserRequest() {
+  return {
+    type: LOGIN_USER_REQUEST
+  }
+}
+
+export function checkToken() {
+  let isToken;
+  sessionStorage.getItem('token')? isToken = true:
+    isToken = false;
```
Comment (4asof4ik) [inline app/src/redux/Login.js]: Read how separate  this operator

File: app/src/components/Secret/Secret.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
```
Comment (4asof4ik) [inline app/src/components/Secret/Secret.js]: Why you add this component

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: app/src/components/Secret/Secret.js (hunk 8 above)
Comment (AlbaMontis) [reply app/src/components/Secret/Secret.js]: I need some component to mount, then token comes. Yulianna will put her component here later.

File: app/src/redux/Login.js (hunk 5 above)
Comment (AlbaMontis) [reply app/src/redux/Login.js]: Done.

File: app/src/redux/Login.js (hunk 7 above)
Comment (AlbaMontis) [reply app/src/redux/Login.js]: Done.

File: app/src/redux/Login.js (hunk 6 above)
Comment (AlbaMontis) [reply app/src/redux/Login.js]: Done.

**Review by denisbodnar — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

File: app/src/redux/Login.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    isToken: isToken
+  }
+}
+
+export function logout() {
+  sessionStorage.removeItem('token');
+  push('/');
+  return {
+    type: LOGOUT_USER
+  }
```
Comment (4asof4ik) [inline app/src/redux/Login.js]: ;

File: app/src/containers/Home/Home.js [hunk 10]
```diff
@@ -1,27 +1,32 @@
...
-          </Col>
-          <Col sm={12} md={4}>
-            <Login />
-            <p className="text-center">OR</p>
-            <Link to='/signup'>
-              <Button bsSize="large" block>SIGN UP</Button>
-            </Link>
-          </Col>
-        </Row>
+        {!this.props.isToken ?
```
Comment (4asof4ik) [inline app/src/containers/Home/Home.js]: Should be IsAuth  and redirect to default page when user is auth

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by yurii-kokhalevych — APPROVED** [review]

**Review by denisbodnar — APPROVED** [review]

**Review by shyyka — APPROVED** [review]

**Review by 4asof4ik — APPROVED** [review]

**Pull request merged**